scss/functions/_get-color.scss

```diff
@@ -7,4 +7,27 @@
   // if they are valid, return the color
   // if they are not valid warn 'Invalid color type or level' and return null
   // null is a special value in Sass that means nothing, and it is used to represent the absence of a value, you need to return null if the type or level is not valid
+
+  // Guard clause for invalid type
+  $type-found: false;
+  @each $key in $color-keys {
+    @if $key == $type {
+      $type-found: true;
+    }
+  }
+  @if $type-found == false {
+    @warn 'Invalid color type: #{$type}';
+    @return null;
+  }
+
+  // Guard clause for invalid level
+  $type-map: map.get($color-box, $type);
+  @if $type-map == null or map.get($type-map, $level) == null {
+    @warn 'Invalid color level: #{$level} for type #{$type}';
+    @return null;
+  }
+
+  // If both type and level are valid, return the color
+  @debug map.get($type-map, $level);
+  @return map.get($type-map, $level);
```

Comment:
sekedar clue kamu cukup menggunakan `@if map.has-key(...) and map.has-key(map.get(...), ...) { ... } @else { ... }`
kamu tinggal mengisi titik titik dari clue yang kamu dapatkan di atas